content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md

```diff
@@ -29,6 +29,14 @@ see the [Creating a cluster with kubeadm](/docs/setup/production-environment/too
 * Swap disabled. You **MUST** disable swap in order for the kubelet to work properly.
     * For example, `sudo swapoff -a` will disable swapping temporarily. To make this change persistent across reboots, make sure swap is disabled in config files like `/etc/fstab`, `systemd.swap`, depending how it was configured on your system.
 
+{{< note >}}
+The `kubeadm` installation is done via binaries that use dynamic linking and assumes that your target system provides `glibc`.
```

Comment:
can you please explain what binaries are using dynamic linking? here the "via" in particular is confusing to me.

all k8s release artifacts use golang static linking. the only exception was the kubelet at some point, not sure if still true.

there should not be a glibc dependency and yes kubeadm / kubelet / k8s has alpine users.

Reply:
@neolit123, please see @jpetazzo 's [comments](https://github.com/kubernetes/website/issues/39444#issuecomment-1430900719) on issue #39444 for more details. Thanks

Reply:
thanks for the link. i see one problem in that discussion - there is no feedback from a kubelet maintainer.
until that happens i don't think we should merge notes in the kubeadm documentation.

if a kubelet maintainer confirms that the kubelet **must** remain with the dynamic linking to system libs then adding the note makes sense. however, if the dynamic linking is accidental (i.e. coming from a CGO dependency) and is not really needed as hinted here:
https://github.com/kubernetes/website/issues/39444#issuecomment-1429624051
then it means the kubelet must instead resolve the issue.

Reply:
there is also a group under SIG Architecture that deals with similar code organization problems.
see #k8s-code-organization in k8s slack.

Reply:
It makes sense. Thank you for looking on that

Reply:
SIG Release can help with some of the relevant details, too.